hr_employee_seniority/models/hr_employee.py

```diff
@@ -22,29 +22,49 @@ class HrEmployee(models.Model):
         compute='_compute_months_service',
     )
 
-    @api.depends('contract_ids')
+    def _first_contract(self):
+        Contract = self.env['hr.contract']
+        return Contract.search([('employee_id', '=', self.id)],
+                               order='date_start asc', limit=1)
+
+    @api.depends('contract_ids', 'initial_employment_date')
     def _compute_months_service(self):
+        date_now = fields.Date.today()
```

Comment:
@gurneyalex Probaly it's best to have to seniority in days, like this we will avoid calculations in months...